moto/batch/models.py

```diff
@@ -520,8 +521,7 @@ def _container_details(self) -> Dict[str, Any]:
         details["memory"] = self._get_container_property("memory", 512)
         details["volumes"] = self._get_container_property("volumes", [])
         details["environment"] = self._get_container_property("environment", [])
-        if self.log_stream_name:
-            details["logStreamName"] = self.log_stream_name
+        details["logStreamName"] = self.log_stream_name
```

Comment:
This change would result in `logStreamName` always being returned, regardless of the status of the job. But the docs state that `Each container attempt receives a log stream name when they reach the RUNNING status.`

Can we change this to only return the `logStreamName` if the status is in `RUNNING/SUCCEEDED/FAILED`? As far as I can tell, that would fix your problem as well, as `batch_simple` will immediately mark jobs as `SUCCEEDED`.

Reply:
I missed that part in the documentation, my bad.

That would indeed do the trick. However, knowing this now, I think that would introduce unexpected behavior in some use cases. For example, if the job had dependencies that `FAILED`, it would be marked as `FAILED` without ever reaching the `RUNNING` status. In that case, it should not include the `logStreamName`, and we would be including it.

I suggest reverting the current changes, and simply modifying the `batch_simple` model to override the `self.log_stream_name` in the `Job` after initialization. It might look a bit ugly, but this way we ensure the proper behavior in `batch` and still get benefits in `batch_simple`. WDYT?

Additionally, would it make sense to include tests for the `logStreamName` expected behavior in `batch`, or it would be an overkill?

Reply:
> I suggest reverting the current changes, and simply modifying the batch_simple model to override the self.log_stream_name
>

Yep, that would work as well, I'd be OK with that.
This new behaviour would definitely need some tests, to verify we don't accidentally break it later on.

 I believe `batch` currently only tests whether the `logStreamName` exists, nothing else. If you want to extend this, and add more assertions for the current behaviour, that's always welcome!